pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1068,23 +1061,32 @@ public CompletableFuture<Optional<Topic>> getTopic(final TopicName topicName, bo
                                         });
                             }
                             return loadOrCreatePersistentTopic(tpName, createIfMissing, properties, topicPolicies);
-                        }).thenCompose(optionalTopic -> {
-                            if (!optionalTopic.isPresent() && createIfMissing) {
```

Comment:
This code was introduced by #21545, but the problem it not exists because a check to determine if the topic exists is now performed in advance.
https://github.com/apache/pulsar/pull/22860/files#diff-0210356c8a88e4efa89eb769a027fa6c166db479dbad8bbbbc704c6ed6e317f5L1038-L1040

At the time, there was a test named `testClientWithAutoCreationGotNotFoundException`. Despite the removal of these lines, the test still passes.